### PR TITLE
Added skin-deep for better React tests

### PR DESCRIPTION
**Pull request by ruyadorno**
It seems the small snippet from `spec/helpers/createComponent.js` evolved into a properly supported library with much more options that will be useful to allow me to test the prompts. This PR introduces this new lib: [`skin-deep`](https://github.com/glenjamin/skin-deep)

Also, updated dependencies to versions compatible with `react@0.14.7`

@stefanbuck for CR

### CODE REVIEW

File: src/renderer/containers/root.jsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default React.createClass({
   render() {
     return (
       <Provider store={store}>
-        {() => <App />}
+        <App />
```
Comment (ruyadorno) [inline src/renderer/containers/root.jsx]: this is related to updating the other dependencies

File: spec/helpers/createComponent.js [hunk 2]
```diff
@@ -1,11 +1,7 @@
-var React = require('react/addons');
-var TestUtils = React.addons.TestUtils;
-
-// Thanks for sharing, Simon Smith.
-// https://gist.github.com/simonsmith/b478d6166acd57829d15
```
Comment (ruyadorno) [inline spec/helpers/createComponent.js]: thanks again @simonsmith for leaving the comment on the snippet about `skin-deep` that saved quite some time :wink:

Comment (simonsmith) [inline spec/helpers/createComponent.js]: :+1:

**Comment by stefanbuck** [comment]
Nice find @ruyadorno. Overall looks good, just question. Why did you changed the patch version for some dependencies?  Shouldn't this be handled by npm and the semver range pattern?

**Comment by ruyadorno** [comment]
for no reason whatsoever, I was just browsing the react-related dependencies and bumping up to latest versions but it happened that I also bumped the patch versions...

it's to avoid this kind of confusion that I tried to use [x instead of patch versions in the past](https://github.com/yeoman/yeoman-app/blob/6b7914951150a6777575d1203f30022a9799ae5b/package.json#L19) - anyways, do you want me to revert it? in this case we should rather standardize the use of `0` or `x` or `*` for the patch versions :blush:

**Comment by stefanbuck** [comment]
All good, thanks for clarification

**Pull request merged**